api/v1beta2/foundationdbcluster_types.go

```diff
@@ -576,15 +568,14 @@ func (processGroupStatus *ProcessGroupStatus) addCondition(oldProcessGroups []*P
 		}
 	}
 
-	// Check if we got a condition for the condition type for the ProcessGroupStatus
-	if oldProcessGroupStatus != nil {
-		for _, condition := range oldProcessGroupStatus.ProcessGroupConditions {
-			if condition.ProcessGroupConditionType == conditionType {
-				// Assumption: processGroupStatus doesn't have the conditionType
-				processGroupStatus.ProcessGroupConditions = append(processGroupStatus.ProcessGroupConditions, condition)
-				return
-			}
+	// If a process group is marked for removal and is fully excluded we only keep the ResourcesTerminating condition.
+	if processGroupStatus.IsMarkedForRemoval() && processGroupStatus.IsExcluded() {
+		if conditionType != ResourcesTerminating {
+			return
 		}
+
+		// Reset all previous conditions
```

Comment:
This is a bit odd. The user is asking a condition to be set, instead this function is clear all existing conditions. Maybe an argument needs to be added so user can specify whether to clear these conditions or not? And, if clearing these conditions is important why is the user not explicitly clearing them, instead of doing that as a side-effect of calling this function?

Reply:
Good point and I agree with the statement that this side-effect might be surprising. I will create a dedicated method to clean conditions if a process group is marked for removal 👍